.gitignore

```diff
@@ -0,0 +1 @@
+*.xpkg
```

Comment:
might be wise to add a ignore for `crossplane-azure-provider-key.json` too, to help avoid that file getting accidentally committed

Reply:
If we all agree, I will make this change across all platform-ref repos.

Reply:
i'd go ahead and include this ignore in this PR since it's a simple thing while we're in here with no real regression risk or testing reset.  Feel free to take a look to see if something is applicable to the other repos - it may not be depending on their instructions and existing .gitignores.